src/core/tsi/s2a/handshaker/s2a_handshaker_client.h

```diff
@@ -111,16 +111,43 @@ class S2AHandshakerClient {
    *  no further handshaker requests will be scheduled with the S2A. **/
   void Shutdown();
 
-  /** This method is exposed for testing purposes only. **/
-  grpc_byte_buffer* get_send_buffer_for_testing();
+  void MaybeCompleteTsiNext(
+      bool receive_status_finished,
+      s2a_recv_message_result* pending_recv_message_result);
+
+  /** This method dereferences all of the data owned by this S2A handshaker
```

Comment:
The comment is confusing a little bit. I think it dereferences a handshaker client and clean up all of its fields once the ref count is zero.

Reply:
I've made this comment more precise in the next commit. PTAL.